drivers/wifi/nrf700x/osal/os_if/inc/osal_structs.h

```diff
@@ -46,6 +46,23 @@ enum wifi_nrf_osal_dma_dir {
 };
 
 
+#ifdef CONFIG_NRF_WIFI_LOW_POWER
+/**
+ * enum wifi_nrf_ps_state - Indicates current RPU Power save state.
+ * @WIFI_NRF_RPU_PS_STATE_SLEEP: RPU is in Sleep state.
+ * @WIFI_NRF_RPU_PS_STATE_AWAKE: RPU is in Awake state.
+ * @WIFI_NRF_RPU_PS_STATE_MAX: Invalid RPU power save state.
+ *
+ * This enum lists the possible RPU power save states.
+ */
+enum wifi_nrf_rpu_ps_state {
+	WIFI_NRF_RPU_PS_STATE_ASLEEP,
+	WIFI_NRF_RPU_PS_STATE_AWAKE,
+	WIFI_NRF_RPU_PS_STATE_MAX
+};
+#endif /* CONFIG_NRF_WIFI_LOW_POWER */
+
```

Comment:
Move this to a FW shared header file (rpu_if.h?) rather than a osal header file.